Assets/PresetsPerFolder/PresetPostProcessor.cs

```diff
@@ -0,0 +1,219 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Experimental;
+using UnityEditor.Presets;
+using UnityEngine;
+
+namespace PresetsPerFolder
+{
+    /// <summary>
+    /// This class is applying Presets assets automatically to each assets added to the folder containing the Preset or any subfolder.
+    /// The code is divided in 3 parts that setup importer dependencies to make sure the importer of any asset stay determinist.
```

Comment:
Suggestion: "The code is divided into 3 parts that setup importer dependencies to make sure the importers of all assets stay deterministic."